docs/api-testing/use-cases/set-variable.md

```diff
@@ -0,0 +1,261 @@
+
+It will return the following object:
+
+```json
+{
+  "id": 3,
+  "name": "Bluetooth Headphones",
+  "price": 49.99,
+  "category": "1",
+  "description": "This is product 3!",
+  "quantity": 50,
+  "imageURL": "http://image.com",
+  "color": [
+    "green",
+    "yellow"
+  ],
+  "createdAt": "2021-11-28T21:58:43.462Z",
+  "updatedAt": "2021-11-28T21:58:43.462Z"
+}
+```
+
+## Mode: Language
+
+This mode is the most advanced way to create your variables. The available options are: `Javascript` and `Template`.
+
+#### Lang: Javascript
+
+In this mode you can create your variable by writing a JavaScript script in the `Body` field. It can be a complete script with variable declarations or loops.
+
+For example, you have a JSON Web Token (JWT) token stored in the `token` variable and you need to decode it and return the JSON payload it was generated from:
+
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2NTgyMzY1NjgsImV4cCI6MTY4OTc3MjU2OCwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoiam9obi5kb2VAZXhhbXBsZS5jb20iLCJOYW1lIjoiSm9obiIsIlN1cm5hbWUiOiJEb2UiLCJFbWFpbCI6ImpvaG4uZG9lQGV4YW1wbGUuY29tIiwiUm9sZSI6WyJNYW5hZ2VyIiwiUHJvamVjdCBBZG1pbmlzdHJhdG9yIl19.DN7vKPlHkAy1hwYOYpUKDwkV0yD-KS2pdoc76aKPhm8
+```
+
+To do so, you need to write the following script inside the `Body` field:
+
+```javascript
+var pieces = token.split('.')
+var b64payload = pieces[1]
+var decoded = Buffer.from(b64payload, 'base64').toString()
+var json = JSON.parse(decoded)
+return json
+```
+
+<img src={useBaseUrl('img/api-testing/set-javascript.png')} alt="Set javascript"/>
+
+That will produce the following JSON:
+
+```json
+{
+  "iat": 1658236568,
+  "exp": 1689772568,
+  "aud": "www.example.com",
+  "sub": "john.doe@example.com",
+  "Name": "John",
+  "Surname": "Doe",
+  "Email": "john.doe@example.com",
+  "Role": [
+    "Manager",
+    "Project Administrator"
+  ]
+}
+```
+
+Then, you can retrieve all the keys as `jsonData.iat` where `jsonData` is the variable name you entered in the `Variable` field.
+
+#### Lang: Template
+
+In this mode, you can create your own template in the same way as it is done for Request Body, the advantage here is that you can print the variable to check if all is correct (Body cannot be printed).
+
+For example, if you need to add a new product to your database, you can create the body for the (PUT) request and paste it in the `Body` field, and then print it in a `Comment`.
+
+```json
+{
+    "id": 4,
+    "name": "T-Shirt",
+    "price": ${price},
+    "category": "1",
+    "description": "This is product ${id}!",
+    "quantity": 5,
+    "imageURL": "http://image.com",
+    "color": ["red", "green"],
+    "createdAt": "${D.format (D.nowMillis(), 'yyyy-MM-DD')}",
+    "updatedAt": "${D.format (D.nowMillis(), 'yyyy-MM-DD')}T${D.format(D.nowMillis(), 'HH:mm:ssz')}"
+}
+```
+
+<img src={useBaseUrl('img/api-testing/set-template.png')} alt="Set template"/>
+<img src={useBaseUrl('img/api-testing/set-template-comment.png')} alt="Comment" width="700"/>
```

Comment:
general comment for all screen shots: To be consistent with the rest of the screen shots, it would look better to grab with whole module in the images instead of having it cut off on the right.    As in the previous image. This can be fixed in a later PR. 
```suggestion
<img src={useBaseUrl('img/api-testing/set-template-comment.png')} alt="Comment" width="700"/>
```